### PR TITLE
irc: implement StreamerQueue for .q(ueue)

### PR DESCRIPTION
**Pull request by resttime**
Attempts to mirror .q(ueue) given by Hanyuu-sama 

https://github.com/R-a-dio/Hanyuu-sama/blob/657a5afa4e9e79e45c2e2b0b37eafa1ea657ed0b/bot/hanyuu_commands.py#L148

For #19

### CODE REVIEW

File: ircbot/commands_impl.go [hunk 1]
```diff
@@ -69,7 +69,50 @@ func LastPlayed(e Event) error {
...
+func StreamerQueue(e Event) error       {
+	// Get queue from streamer
+	songQueue, err := e.Bot.Streamer.Queue(e.Context())
+	if err != nil {
+		return err
+	}
+
+	// If the queue is empty then we're done
+	if len(songQueue) == 0 {
+		e.EchoPublic("No queue at the moment")
```
Comment (Wessie) [inline ircbot/commands_impl.go]: you can use `NewPublicError(nil, "No queue at the moment")` here to get an error that the handler knows to send to the channel

File: ircbot/commands_impl.go [hunk 2]
```diff
@@ -69,7 +69,50 @@ func LastPlayed(e Event) error {
...
+
+	// Define the message strings
+	message := "{green}Queue (/r/ time: %s):{clear} %s"
+	messageJoin := " {red}|{clear} "
+
+	// Grab metadata and color green if requestable
+	onlyMetadata := make([]string, len(songQueue))
+	for i, song := range songQueue {
+		if song.IsUserRequest {
+			onlyMetadata[i] = "{green}" + song.Metadata + "{clear}"
```
Comment (Wessie) [inline ircbot/commands_impl.go]: User supplied data shouldn't be inserted directly into the output message like this, the way our formatting works is two-step; it first does all the IRC color coding (`{green}`, `{clear}`, etc) and then it does the normal formatting (`%s`, `%d`, etc). So if you don't use these a song that contains any of the color codings or even the normal formatting will cause unexpected output. So all user output should be given as argument to the formatting functions, either `Fmt` or one of the `Echo`'s

File: ircbot/commands_impl.go [hunk 3]
```diff
@@ -69,7 +69,50 @@ func LastPlayed(e Event) error {
...
+		}
+	}
+
+	// Format the message
+	message = fmt.Sprintf(message,
+		FormatPlaybackDurationHours(totalQueueTime),
+		strings.Join(onlyMetadata, messageJoin))
+
+	// Echo out the message
+	e.EchoPublic(message)
```
Comment (Wessie) [inline ircbot/commands_impl.go]: the `Echo` functions do formatting for you, so the above `fmt.Sprintf` is unneeded and you can pass them directly to `EchoPublic` instead

**Comment by resttime** [comment]
Gonna resolve these conflicts, time for git magic

**Comment by resttime** [comment]
THAT should do it now

**Comment by Wessie** [comment]
looks good, thank you

**Pull request merged**